runtime/domain.c

```diff
@@ -935,13 +936,29 @@ void caml_update_minor_heap_max(uintnat requested_wsz) {
   check_minor_heap();
 }
 
-void caml_init_domains(uintnat minor_heap_wsz) {
+void caml_init_domains(uintnat minor_heap_wsz, uintnat max_domains) {
   int i;
 
+  all_domains = caml_stat_alloc_noexc(max_domains * sizeof(dom_internal));
+  if (all_domains == NULL)
+    caml_fatal_error("Failed to allocate all_domains");
+
+  stw_request.participating =
+      caml_stat_alloc_noexc(max_domains * sizeof(dom_internal*));
+  if (stw_request.participating == NULL)
+    caml_fatal_error("Failed to allocate stw_request.participating");
+
+  stw_domains.domains =
+      caml_stat_alloc_noexc(max_domains * sizeof(dom_internal*));
+  if (stw_domains.domains == NULL)
+    caml_fatal_error("Failed to allocate stw_domains.domains");
+
+  caml_init_gc_stats(max_domains);
```

Comment:
I think this call should probably move to `caml_init_gc`, as it initializes a runtime service that is independent from the domains machinery. (I have no idea what `caml_init_signal_handling` is doing at the end of the current function, and the same comment would apply -- or maybe even move it to `caml_start_*` -- but oh well.)

Reply:
Done in https://github.com/ocaml/ocaml/pull/13272/commits/263be361baad53f6ef36f8774ac8b5d9b1572525.